### ISSUE TITLE
Missing Drag-n-drop functionality in vertical file switcher 

### ISSUE THREAD

**Issue by kamilrymarz** [notepad-plus-plus/notepad-plus-plus#2221]
Hi,
I really like N++ and I've been using it for last couple of years. Now i figured out that it would be great to help you guys with making this tool  even better. I found some missing feature and if you accept it I could try to implement it.
### Description of the Issue

Currently there is no possibility to rearrange lists elements in vertical file switcher. It would be very useful to have a possibility to rearrange all items in any order by drag-n-drop operation.

### Expected Behavior

A possibility to rearrange items in vertical file switcher by drag-n-drop operation.

### Actual Behavior

Actual user only can sort items by name.

I am wondering which implementation would be better:
- elements are switched immediately during dragging
- new place for dragged image is marked during dragging and element is moved only once when left button is released
For the begging I could start with rearranging items only, later I could add functionality to open dragged files in new window and all other features that are implemented for horizontal tab bar.

If someone accept this then I'll create pull request with this feature.

**Comment by patuh9090** [comment]
Good idea, I think the best would be the first best samples. Thus, this function is arranged in Sublime text3, I use it for about a year, very convenient.

**Comment by A-R-C-A** [comment]
I can not guaranty that this will be accepted, but it would be an improvement IMO. I'd prefer the second implementation, but it's finally up to you.

Some more ideas:
* Shifting Doc Switcher from preference dialog to 'View/Doc Switcher' and eventually in the toolbar.
* Adding more columns to Doc Switcher, at least 'Date Modified' and making it optional like the extension column.

**Comment by kamilrymarz** [comment]
Thanks for comments.

After few tests I also think that second implementation is better. I'll create PR when it will be finished.

Shifting Doc Switcher is also a good idea.  Columns to display could be chosen from a list in a new dialog which will pop up after right click on bar with column's name.

**Comment by robotboy655** [comment]
Can we please get a status update on this? Why was https://github.com/notepad-plus-plus/notepad-plus-plus/pull/2243 closed with a comment saying "This feature has been implemented.", but drag'n'drop still doesn't work for Document List panel in 2023?

**Comment by peter-galcik** [comment]
> Can we please get a status update on this? Why was #2243 closed with a comment saying "This feature has been implemented.", but drag'n'drop still doesn't work for Document List panel in 2023?

Yes, also add vote to this issue. Its very pain in the ass that we don't have this functionality, which I  need to use frequently when  tidying document and grouping related files  in order.

**Comment by e-motiv** [comment]
Just FYI, although this is not drag and drop the document list follows the tab bar and even if the tab bar is hidden you can still **use the shortcuts from the tab bar to position files in the document list** (down, up, start, end, ..)
